### ISSUE TITLE
PR #578 broke CMAKE_PREFIX_PATH

### ISSUE THREAD

**Issue by Rayman** [catkin/catkin_tools#583]
PR #578 broke CMAKE_PREFIX_PATH. I've narrowed it down to a few lines in a docker container. Without this commit, everything works as expected.

### System Info

* Operating System: Linux 9164750fc67d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* Python Version: 3.6.8
* Version of catkin_tools: 5b121260a75ffcda863d4588a8e9efc7b4945477
* ROS Distro: melodic

### Build / Run Issue

[//]: # (If you are migrating from catkin_make, please follow the migration guide:)
[//]: # (http://catkin-tools.readthedocs.org/en/latest/migration.html)

[//]: # (Please also check for solved issues here:)
[//]: # (http://catkin-tools.readthedocs.org/en/latest/troubleshooting.html)

[//]: # (And check for open issues here:)
[//]: # (https://github.com/catkin/catkin_tools/issues?q=is%3Aopen+is%3Aissue+label%3Abug)

[//]: # (If you have an issue building and running code, please verify the following:)

* [x] Works with `catkin_make`
* [x] Works with `catkin_make_isolated --merge`
* [x] Works with `catkin build`
* [x] Works with `catkin build -p1`

### Expected Behavior
Build a normal workspace

### Actual Behavior
See the error log below

### Steps to Reproduce the Issue

```bash
docker run -it ros:melodic-ros-base
sudo apt update && apt install -y python3-pip
pip3 install git+https://github.com/catkin/catkin_tools.git@5b121260a75ffcda863d4588a8e9efc7b4945477
cd ~/
catkin init
catkin config --extend /opt/ros/melodic --mkdirs
cd src
git clone https://github.com/Rayman/toypkg.git
catkin build
```
### Error log
```
_____________________________________________________________________________________
Errors     << catkin_tools_prebuild:cmake /root/logs/catkin_tools_prebuild/build.cmake.001.log
Not searching for unused variables given on the command line.
Re-run cmake no build system arguments
CMake Error at /root/build/catkin_tools_prebuild/CMakeLists.txt:12 (message):
  The catkin CMake module was not found, but it is required to build a linked
  workspace.  To resolve this, please do one of the following, and try
  building again.



   1. Source the setup.sh file from an existing catkin workspace:
      source SETUP_FILE



   2. Extend another catkin workspace's result (install or devel) space:
      catkin config --extend RESULT_SPACE



   3. Set `catkin_DIR` to the directory containing `catkin-config.cmake`:
      catkin config --cmake-args -Dcatkin_DIR=CATKIN_CMAKE_CONFIG_PATH



   4. Add the catkin source package to your workspace's source space:
      cd SOURCE_SPACE && git clone https://github.com/ros/catkin.git


-- Configuring incomplete, errors occurred!
See also "/root/build/catkin_tools_prebuild/CMakeFiles/CMakeOutput.log".
cd /root/build/catkin_tools_prebuild; catkin build --get-env catkin_tools_prebuild | catkin env -si  /usr/bin/cmake /root/build/catkin_tools_prebuild --no-warn-unused-cli -DCATKIN_DEVEL_PREFIX=/root/devel/.private/catkin_tools_prebuild -DCMAKE_INSTALL_PREFIX=/root/install; cd -
.....................................................................................
Failed     << catkin_tools_prebuild:cmake          [ Exited with code 1 ]            
Failed    <<< catkin_tools_prebuild                [ 0.0 seconds ]                   
Abandoned <<< toypkg
```

**Comment by mvukov** [comment]
👍 Thanks for reporting. I spent a couple of hours on this. Besides a wrong CMAKE_PREFIX_PATH, possibly the same PR influenced very weird handling of env. variables: enclosing env. variables in double quotes. For instance, if I would set in a script
```bash
export FOO=bar
```
and in CMake use
```cmake
set(foo $ENV{FOO})
```
cmake would read this as `"bar"`, which would make a total mess.

Fortunately, rolling back to an older commit solves the issues.

**Comment by mikepurvis** [comment]
@timonegk Looks like there are some unexpected side effects and #578 will have to be rolled back.

**Comment by timonegk** [comment]
I am sorry for causing the inconvenience! I did not see any problem while testing the fix on my machine, but we should have been alerted by the failing CI. Please revert #578 until an actual fix for #528 is found.

**Comment by Rayman** [comment]
Was fixed by #584
